lib/Dialect/Torch/Transforms/DecomposeComplexOps.cpp

```diff
@@ -2845,6 +2845,63 @@ class DecomposeAtenIndexTensorHackedTwinOp
 };
 } // namespace
 
+namespace {
+class DecomposeAtenMseLossOp : public OpRewritePattern<AtenMseLossOp> {
+public:
+  using OpRewritePattern::OpRewritePattern;
+  LogicalResult matchAndRewrite(AtenMseLossOp op,
+                                PatternRewriter &rewriter) const override {
+
+    // The `reduction` arg would have only three valid values.
+    // 0 means no reduction.
+    // 1 means mean reduction.
+    // 2 means sum reduction.
+    int64_t reductionType;
+    if (!matchPattern(op.reduction(), m_TorchConstantInt(&reductionType)))
+      return rewriter.notifyMatchFailure(
+          op, "Expected a constant integer value for reduction");
+
+    Location loc = op.getLoc();
+    BaseTensorType inputType = op.self().getType().cast<BaseTensorType>();
+    BaseTensorType resultType = op.getType().cast<BaseTensorType>();
+    BaseTensorType subType =
+        inputType
+            .getWithSizesAndDtype(
+                !inputType.hasSizes()
+                    ? Optional<ArrayRef<int64_t>>()
+                    : llvm::makeArrayRef(inputType.getSizes()),
+                resultType.getDtype())
+            .cast<BaseTensorType>();
+
+    Value sub = createTensorSub(rewriter, loc, subType, op.self(), op.target());
+    Value result = rewriter.create<AtenSquareOp>(loc, subType, sub);
+
+    if (reductionType >= 1) {
+      int64_t inputRank = getTensorRank(op.self());
+      SmallVector<Value> dimListElements;
+      for (int64_t i = 0; i < inputRank; i++)
+        dimListElements.push_back(rewriter.create<Torch::ConstantIntOp>(
+            loc, rewriter.getI64IntegerAttr(i)));
+      Value dimList = rewriter.create<PrimListConstructOp>(
+          loc, Torch::ListType::get(Torch::IntType::get(op->getContext())),
+          dimListElements);
```

Comment:
I don't think there is a need for this. Both `AtenMeanDimOp` and `AtenSumDimIntListOp` should reduce along all dimensions when `dim=None`